### ISSUE TITLE
Provide `module` in `package.json` along with es6 format Rollup

### ISSUE THREAD

**Issue by brettz9** [qunitjs/qunit#1281]
## Tell us about your runtime:

* **QUnit version**: 2.6.1
* **What environment are you running QUnit in? (e.g., browser, Node)**: Browser (Chrome 66, Firefox 61.0b7)
* **How are you running QUnit? (e.g., script, testem, Grunt)**: script

## What are you trying to do?

Code that reproduces the problem:

```js
import QUnit from './node_modules/qunit/dist/qunit-es6.js';
```
(This is a feature request; I realize this will not work currently.)

If you have any relevant configuration information, please include that here:

## What did you expect to happen?

QUnit could be imported.

In cases where one runs Rollup (and node-resolve with optional commonjs plugins) on one's QUnit tests themselves, the following

```js
import QUnit from 'qunit';
```

...should also work if `module` in `package.json` points to the ES6 modules file:

```json
"module": "dist/qunit-es6.js"
```

## What actually happened?

Nothing, as a module export is not available.

**Comment by villander** [comment]
@trentmwillis how run qunit on rollup? as testing framework?

**Comment by brettz9** [comment]
@villander : Hope you don't mind me interjecting; the idea is twofold...

1) Provide an ES6 Modules export (adding a `format: "es"` bundle in your Rollup) so that test environments (currently modern browsers only, though potentially Node in the future as well) can directly and modularly import QUnit with no need to define their own Rollup routine against Qunit and no need for their code to define globals or add extra script tags to their HTML--their test file(s) need merely import the QUnit ES module file wherever needed.

2) Ensure that a project using ES6 modules and making bundle files out of those modules (e.g., using Rollup to make these bundles) can auto-discover the starting file within QUnit--the ES6 Modules distribution file it should roll up into its bundle. This is indicated by the `module` field in `package.json`.

It is therefore not to run QUnit against or "on" Rollup, but to allow users of QUnit, including those using Rollup (or other ESM bundlers), to import QUnit without needing to know how QUnit bundles itself internally.

**Comment by villander** [comment]
@brettz9 my problem is that i can't import a ES6 module on my `test.js` file, because thist file has a dom elementes, example: 

```js
//test.js

import { $ } from '../src/helpers/dollar';

```

```js
// dollar.js
export const $ = document.querySelector.bind(document);
```

I receive this error

```
TAP version 13
not ok 1 test/test.js > Failed to load the test file with error:
ReferenceError: document is not defined
    at Object.document (/Users/villander/Projects/paycertify.js/src/helpers/dollar.js:1:18)
    at Module._compile (module.js:652:30)
    at Module._compile (/Users/villander/Projects/paycertify.js/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (module.js:663:10)
    at Object.newLoader [as .js] (/Users/villander/Projects/paycertify.js/node_modules/pirates/lib/index.js:88:7)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
  ---
  message: "should be able to load file"
  severity: failed
  actual: false
  expected: true
  stack:     at Object.<anonymous> (/Users/villander/Projects/paycertify.js/node_modules/qunit/bin/run.js:61:13)
```

**Comment by brettz9** [comment]
Support questions should be asked on a site like Stackoverflow. If you link to the question from here, I should be able to offer an answer.

**Comment by trentmwillis** [comment]
Would it suffice for us to simply ship the `src` files? They're currently in ES6 module format so we could just point the default to [`src/qunit.js`](https://github.com/qunitjs/qunit/blob/master/src/qunit.js).

**Comment by brettz9** [comment]
You could technically do so (and indeed point `module` in `package.json` to `src.qunit.js`), but you would have to add ".js" to all of the import statements or otherwise it wouldn't work in the browser.

But I would probably advise against this...

Any external modules you were importing would also have to be resolvable, such as using `import "./node_modules/some-external-module/index-es.js"` rather than `import "some external-module"` (and I noticed you were using `import global from "global";` which would have to be changed as well).

And besides this ugliness in paths, even if you aren't actually importing any external modules (or have a routine to pull them out of `node_modules`), while pointing to `src` would technically be fine,  if you ever decided you wanted to import external modules requiring a path-aware build routine (e.g., having `import jquery from 'jquery';` in source rather than `import jquery from './node_modules/jquery/dist/jquery.js'`), you'd have to introduce a breaking change for your users, who'd need to switch their import paths from the `src` path to the distribution file path you'd then need to make available (since browsers couldn't keep referencing the `src` code as they don't have a way to understand how to resolve the likes of `import jquery from 'jquery';` and perhaps might never define such a way; even for bundlers looking at `module`, this property is supposed to resolve to a directly usable distribution file, not one requiring users to add config that is aware of your bundling process).

Note that Rollup already has a `format: "es"` option and plugins like `rollup-plugin-re` for swapping out something like your "global" import, so there are tools to facilitate such an effort.

**Comment by trentmwillis** [comment]
All good points. I was simply trying to be lazy and solve the immediate request of being able to use Rollup. But, definitely good to consider future browser use cases.

**Comment by brettz9** [comment]
Well, if it'd help it get out faster, there will probably be breaking changes down the road anyways, so would be worthwhile, but in this case, I don't think it will save you any time since even if you point to `src`, you'll still need to do the mentioned refactoring...

**Comment by Krinkle** [comment]
As I understand it, the original use case has since been solved by https://github.com/qunitjs/qunit/issues/1465.

I'm marking this as declined, as we didn't end up going for distributing QUnit itself in ESM form, but I don't think we need that. Doing so imho would be less flexible and more complicated for end-users to manage.  I believe what we ended up doing is simpler and should work for everyone to run ES5 sources and/or tests with QUnit, both in browsers and on Node.js.

If you run into any issues or limitations, do let us know. Thanks!
